google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java

```diff
@@ -163,6 +164,40 @@ public static BlobSourceOption shouldReturnRawInputStream(boolean shouldReturnRa
       return new BlobSourceOption(UnifiedOpts.returnRawInputStream(shouldReturnRawInputStream));
     }
 
+    /**
+     * Deduplicate any options which are the same parameter. The value which comes last in {@code
+     * os} will be the value included in the return.
+     */
+    @BetaApi
+    public static BlobSourceOption[] dedupe(BlobSourceOption... os) {
```

Comment:
I see these helpers are public; I understand the purpose just not the experience. Please reply with a snippet of how code will look when using one of these?

Reply:
Imagine the case where you have some defaults you want to specify, but allow overriding in specific usage.

```java
final class X {
  private static final BlobWriteOption[] defaultWriteOptions = new BlobWriteOption[] {
      BlobWriteOption.doesNotExist(), 
      BlobWriteOption.userProject("proj")
  };

  public static void main(String[] args) throws Exception {
    try (Storage s = StorageOptions.http().build().getService()) {

      BlobInfo info = BlobInfo.newBuilder("bucket", "object").build();

      try (WriteChannel w = s.writer(
          info,
          BlobWriteOption.dedupe(
              defaultWriteOptions, 
              BlobWriteOption.generationMatch(3)
          )
      )) {
        w.write(ByteBuffer.wrap("abc".getBytes(StandardCharsets.UTF_8)));
      }
    }
  }
}
```